renku/core/commands/workflow.py

```diff
@@ -236,3 +208,108 @@ def compose_workflow_command():
     return (
         Command().command(_group_workflow).require_migration().require_clean().with_database(write=True).with_commit()
     )
+
+
+@inject.autoparams("client")
+def unique_workflow(database: Database, client: LocalClient) -> Dict[str, Plan]:
+    """Map of unique plans in the provenance graph indexed by name."""
+    if not client.has_graph_files():
+        raise errors.OperationError("Dataset provenance is not generated. Run `renku graph generate`.")
+
+    workflows = dict()
+    plans = database.get("plans")
+    for p in plans.values():
+        if p.invalidated_at is None and p.name not in workflows:
+            workflows[p.name] = p
+    return workflows
+
+
+@inject.autoparams()
+def _edit_workflow(
+    name,
+    database: Database,
+    new_name: Optional[str],
+    description: Optional[str],
+    set_params: Dict[str, str],
+    map_params: List[str],
+    rename_params: Dict[str, str],
+    describe_params: Dict[str, str],
+):
+    """Edits a workflow details."""
+    derived_from = _find_workflow(name)
+    workflow = derived_from.derive()
+    if new_name:
+        workflow.name = new_name
+
+    if description:
+        workflow.description = description
+
+    for name, value in set_params.items():
+        for param in workflow.parameters:
+            if param.name == name:
+                param.default_value = value
+                break
+        else:
+            workflow.parameters.append(
+                CommandParameter(default_value=value, id=CommandParameter.generate_id(plan_id=workflow.id), name=name)
+            )
+
+    if len(map_params) and isinstance(workflow, CompositePlan):
+        workflow.set_mappings_from_strings(map_params)
+
+    for name, new_name in rename_params.items():
+        for param in workflow.parameters:
```

Comment:
We should probably check if a user tries to rename a param that doesn't exist and raise an error